IBAnimatable/IBEnum.swift

```diff
@@ -5,17 +5,19 @@
 
 import Foundation
 
+
+
 /**
  A protocol provides extension method for converting `String` into `enum`.
  Because `@IBInspectable` property can not support `enum` directly. To provide both `enum` API in code and `@IBInspectable` supported type `String` in Interface Builder, we use `IBEnum` to bridge Swift `enum` and `String`
  */
-public protocol IBEnum: StringLiteralConvertible {
+public protocol IBEnum {
   /**
    Initializes a swift `enum` with provided optional string
    
    - Parameter string: The optional string to be converted into `enum`.
    */
-  init(string: String?)
+  init?(string: String?)
```

Comment:
I tried something similar too. One inconvenient is that it can lead to duplicate code (for instance if we want to have a default implementation of `IBEnum` and `IBOptionalEnum` for `RawRepresentable` enums). I prefer to have just one enum for this purpose.